os/os.proto

```diff
@@ -0,0 +1,321 @@
+// This file defines a gNOI API used for OS installation.
+syntax = "proto3";
+
+package gnoi.os;
+
+import "github.com/openconfig/gnoi/types/types.proto";
+
+option (types.gnoi_version) = "0.1.0";
+
+// The OS service provides an interface for OS installation on a Target. The
+// Client progresses through 3 RPCs:
+//   1) Installation - provide the Target with the OS package.
+//   2) Activation - activate an installed OS package.
+//   3) Verification - verify that the Activation was successful.
+//
+// Dual Supervisor Target is supported, where the above process is executed once
+// for each Supervisor.
+//
+// Note that certain platforms may have particular approaches to upgrade the
+// firmware of specific components, eg., power supply units, etc.. In addition,
+// platforms may have processes to apply patches to the running OS. Handling
+// these exceptions introduces extra complexities. For Targets that implement
+// this service, component firmware upgrade or OS patching must be embedded
+// within an OS upgrade.
+service OS {
+  // Install transfers an OS package into the Target. No concurrent Install RPCs
+  // must be allowed to the same Target.
+  //
+  // The OS package file format is platform dependent. The platform must
+  // validate that the OS package that is supplied is valid and bootable. This
+  // should include a hash check against a known good hash. It is recommended
+  // that the hash is embedded in the OS package.
+  //
+  // The Target manages its own persistent storage, and OS installation process.
+  // It stores a set of distinct OS packages, and always proactively frees up
+  // space for incoming new OS packages. It is guaranteed that the Target always
```

Comment:
So, to be clear the expectation is that apart from the current running OS version, any other packages may be arbitrarily deleted by starting a transfer request? Just want to make sure that that is the expectation and that the loss of control is ok.

Reply:
Correct. The Operator must not have the expectation that the Target is keeping every installed package besides the last installed one, or the  active one.

Reply:
Correct. However, (and this is up to the platform implementation) if the size of the storage is orders of magnitude larger than the typical size of the package, there is value in keeping a buffer of packages. The logic for removing installed packages could for example be FIFO. All of this is up to the platform  implementer and out of the scope of this service. All the Client must expect is that the last Installed package is always available. I will make this clearer in the wording.